cpp/util/SnowflakeCommon.cpp

```diff
@@ -10,10 +10,13 @@
 #include "snowflake/platform.h"
 #include "snowflake/Proxy.hpp"
 #include "../logger/SFLogger.hpp"
+#include <stdexcept>
 
 using namespace Snowflake;
 using namespace Snowflake::Client;
 
+static bool exit_on_memory_error = true;
```

Comment:
I would refer the name of `exception_on_memory_error`. exit is not good and it could be fixed in the future for C API as well, while it won't use exception.